datastore/src/main/kotlin/com/openlattice/collaborations/controllers/CollaborationController.kt

```diff
@@ -0,0 +1,189 @@
+import com.openlattice.collaborations.CollaborationsApi.Companion.TABLE_ID
+import com.openlattice.collaborations.CollaborationsApi.Companion.TABLE_ID_PATH_PARAM
+import com.openlattice.organizations.OrganizationDatabase
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.*
+import java.util.*
+import java.util.stream.Collectors
+import javax.inject.Inject
+
+@RestController
+@RequestMapping(CONTROLLER)
+@SuppressFBWarnings(
+        value = ["RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE"],
+        justification = "Allowing redundant kotlin null check on lateinit variables")
+class CollaborationController : AuthorizingComponent, CollaborationsApi {
+
+    @Inject
+    private lateinit var authorizations: AuthorizationManager
+
+    @Inject
+    private lateinit var collaborationService: CollaborationService
+
+    @Timed
+    @GetMapping(value = ["", "/"], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun getCollaborations(): Iterable<Collaboration> {
+        val authorizedCollaborationIds = getAllAuthorizedCollaborationIds()
+        return collaborationService.getCollaborations(authorizedCollaborationIds).values
+    }
+
+    @Timed
+    @PostMapping(value = ["", "/"], consumes = [MediaType.APPLICATION_JSON_VALUE], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun createCollaboration(@RequestBody collaboration: Collaboration): UUID {
+        return collaborationService.createCollaboration(collaboration, Principals.getCurrentUser())
+    }
+
+    @Timed
+    @GetMapping(value = [ID_PATH_PARAM], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun getCollaboration(@PathVariable(ID) id: UUID): Collaboration {
+        ensureReadAccess(AclKey(id))
+        return collaborationService.getCollaboration(id)
+    }
+
+    @Timed
+    @GetMapping(value = [ORGANIZATIONS_PATH + ORGANIZATION_ID_PATH_PARAM], produces = [MediaType.APPLICATION_JSON_VALUE])
+    override fun getCollaborationsIncludingOrganization(@PathVariable(ORGANIZATION_ID) organizationId: UUID): Iterable<Collaboration> {
+        ensureReadAccess(AclKey(organizationId))
+        val collaborations = collaborationService.getCollaborationsIncludingOrg(organizationId)
+
+        val authorizedIds = filterToAuthorizedIds(collaborations.map { it.id })
+
+        return collaborations.filter { authorizedIds.contains(it.id) }
+    }
+
+    @Timed
+    @DeleteMapping(value = [ID_PATH_PARAM])
+    override fun deleteCollaboration(@PathVariable(ID) id: UUID) {
+        ensureOwnerAccess(AclKey(id))
+        collaborationService.deleteCollaboration(id)
+    }
+
+    @Timed
+    @PostMapping(value = [ID_PATH_PARAM + ORGANIZATIONS_PATH], consumes = [MediaType.APPLICATION_JSON_VALUE])
+    override fun addOrganizationIdsToCollaboration(@PathVariable(ID) id: UUID, @RequestBody organizationIds: Set<UUID>) {
+        ensureOwnerAccess(AclKey(id))
```

Comment:
What about `ensureReadAccess` on the org ids? Or even going as far as ensuring the caller is a member of the orgs?

Reply:
I don't think that's necessary as we don't auth check org UUIDs anywhere else. We can revisit whether we want to hide ids and do a pass over our codebase at some point, but we've had lengthy conversations about this in the past that have resulted in our current model.